tests/MongoDB.Driver.GridFS.Tests/Specifications/gridfs/tests/upload.yml

```diff
@@ -156,3 +156,18 @@ tests:
                     { insert : "expected.chunks", documents : [
                         { _id : "*actual", files_id : "*result", n : 0, data : { $hex : "11" } }
                     ] }
+    -
+        description: "Upload when length is 0 sans MD5"
```

Comment:
I know I already lgtm'ed this pull request, but I realized after the fact that we should probably add one more test just like this one but with a non-zero length source.

Reply:
@jyemin can you comment on this?

Perhaps we just need to get this committed ASAP.

Reply:
Will it exercise additional code paths?

Reply:
Yes. It would have revealed the NullReferenceException bug that I caught visually in an earlier commit.

Reply:
I think a test should be added.  Shouldn't take long.